browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -21,14 +21,36 @@
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_finder.h"
 #include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/browser_list_observer.h"
 #include "chrome/browser/ui/tabs/tab_strip_model.h"
 #endif
 
 namespace brave_rewards {
 
+#if !BUILDFLAG(IS_ANDROID)
+class BraveBrowserListObserver : public BrowserListObserver {
+ public:
+  explicit BraveBrowserListObserver(RewardsTabHelper* tab_helper)
+      : tab_helper_(tab_helper) {}
+  ~BraveBrowserListObserver() override {}
+  void OnBrowserSetLastActive(Browser* browser) override {
+    tab_helper_->OnBrowserSetLastActive(browser);
+  }
+  void OnBrowserNoLongerActive(Browser* browser) override {
+    tab_helper_->OnBrowserNoLongerActive(browser);
+  }
+
+ private:
+  raw_ptr<RewardsTabHelper> tab_helper_;  // Not owned.
```

Comment:
`const` all constructor-initialized pointers to trigger compile errors if initialization is missing.